sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -110,6 +123,15 @@ static event_queue_t _queue;
 static uint8_t _listen_buf[CONFIG_GCOAP_PDU_BUF_SIZE];
 static sock_udp_t _sock_udp;
 
+#ifdef CONFIG_GCOAP_ENABLE_DTLS
+    #define SOCK_DTLS_CLIENT_TAG (2)
+    static sock_dtls_t _sock_dtls;
+    static kernel_pid_t _main_thread;
+
+    static event_timeout_t _dtls_session_free_up_tmout;
+    static event_callback_t _dtls_session_free_up_tmout_cb;
+#endif
```

Comment:
Here too

Reply:
I think you can even drop the preprocessor `#if` around here and use `if (IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS))` through-out the code. This way we get rid of a lot of pre-processor magic, which allows you to not hide your code from the compiler in any case.